csrc/scheduler/utils.cpp

```diff
@@ -1216,6 +1216,24 @@ std::vector<TensorView*> cacheInputs(Fusion* fusion, bool unroll) {
     for (auto use : tv->uses()) {
       if (!use->isA<PadOp>()) {
         cached_uses.push_back(use);
+      } else {
+        // propagate allocation domain on pad op
+        if (!use->as<PadOp>()->out()->isA<TensorView>()) {
+          continue;
+        }
+        auto pad_out_tv = use->as<PadOp>()->out()->as<TensorView>();
+        if (pad_out_tv->hasAllocation()) {
+          continue;
+        }
+        std::optional<std::vector<int64_t>> alloc_permutation =
+            ir_utils::computePermutation(
+                tv->getLogicalDomain(), tv->getMaybeAllocationDomain());
+        if (alloc_permutation.has_value()) {
+          pad_out_tv->setAllocationDomain(
```

Comment:
I understand that this is good enough for the validation issue, and I don't have any alternative suggestion at this moment, but setting allocation domains of intermediate tensors makes me a little concerned as this allocation domain isn't guaranteed to be the correct allocation domain. Since it isn't a global memory tensor, for example, if it gets split by a non-divisible factor, its allocation size would need to be larger than the logical domain.

IIRC, the allocation pass of the lowering doesn't actually use this set allocation domain, so potentially incorrect allocation domains won't matter, but that itself should be fixed as well.